src/test/java/nextstep/subway/line/LineAcceptanceTest.java

```diff
@@ -1,72 +1,158 @@
 package nextstep.subway.line;
 
+import io.restassured.RestAssured;
+import io.restassured.response.ExtractableResponse;
+import io.restassured.response.Response;
 import nextstep.subway.AcceptanceTest;
+import nextstep.subway.line.dto.LineRequest;
+import nextstep.subway.line.dto.LineResponse;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
 
 @DisplayName("지하철 노선 관련 기능")
 public class LineAcceptanceTest extends AcceptanceTest {
-    @DisplayName("지하철 노선을 생성한다.")
+
     @Test
+    @DisplayName("지하철 노선을 생성한다.")
     void createLine() {
-        // when
-        // 지하철_노선_생성_요청
+        // given - 지하철_노선_생성_요청
+        LineRequest lineRequestOfShinBundang = new LineRequest("신분당선", "bg-red-600");
```

Comment:
요청 파라미터를 생성하는 것이 인수 테스트에서의 given 조건에 들어가는지 한 번 생각해보시면 좋을 것 같습니다 :)

Reply:
음 이부분은 저도 하면서 좀 모호했던 부분이에요.
지하철 노선 생성 요청의 역할에서

``` java
private static final String LINE_NAME_SHIN_BUN_DANG = "신분당선";
private static final String LINE_COLOR_SHIN_BUN_DANG = "bg-red-600";

response = 지하철_노선_생성_요청(LINE_NAME_SHIN_BUN_DANG, LINE_COLOR_SHIN_BUN_DANG);

private ExtractableResponse<Response> 지하철_노선_생성_요청(String name, String color) {
        return RestAssured
                .given().log().all()
                .body(new LineRequest(name, color))
                .contentType(MediaType.APPLICATION_JSON_VALUE)
                .accept(MediaType.APPLICATION_JSON_VALUE)
                .post("/lines")
                .then().log().all()
                .extract();
    }
```

이러한 형태로 노선의 이름과 색상을 외부에서 주입을 받게 했습니다.

그리고 테스트 시 "신분당선", "bg-red-600" 의 하드코딩이 반복되어 해당 내용은 상수로 분리했습니다.
그런데 프로덕션 코드에서는 반복되는 상수를 분리하는 방법이 좋다고 생각을 하는데,
테스트 코드에서는 각각의 시나리오의 테스트 코드를 작성하게 되는데 하드코딩 상수화가 가독성을 해치는건 아닐까 ? 라는 고민이 있었어요.
영철님의 생각을 여쭤보고 싶어요 !

Reply:
> 그리고 테스트 시 "신분당선", "bg-red-600" 의 하드코딩이 반복되어 해당 내용은 상수로 분리했습니다.
그런데 프로덕션 코드에서는 반복되는 상수를 분리하는 방법이 좋다고 생각을 하는데,
테스트 코드에서는 각각의 시나리오의 테스트 코드를 작성하게 되는데 하드코딩 상수화가 가독성을 해치는건 아닐까 ? 라는 고민이 있었어요.

답은 결국 `상황에 맞게` 인 것 같습니다!
하드 코딩으로 했다가 테스트 코드의 유지, 보수의 목적으로 상수화를 시켰다가 불필요하게 테스트 간의 커플링이 생길 경우 이를 분리할 수도 있을 것 같습니다.

저는 지금과 같은 경우는 하드 코딩으로 하는 편인데 그렇게해도 유지보수에 있어서 큰 불편함이 없었고 직접 값을 넣는게 테스트 시에는 오히려 읽히기 쉬운 경우가 있더라구요 :)
그리고 자주 혹은 다른 테스트에 공유되어 사용되는 데이터(주로 test fixture)인 경우 상수화를 하는 것이 좋다고 생각하는 편입니다!

개인마다 특정 경험을 통해 느끼는 바가 다르기 때문에 이번 과정을 통해 여러가지 고민해보시고 시도해보시고 같이 좋은 의견도 나누면서
한결님 나름의 노하우가 쌓이고 이를 실무에도 적용해보시면 best일 것 같습니다 👍 

---

단위 테스트에서는 함수/메서드에 사용될 파라미터를 given으로 주로 사용하지만
```java
// given
final int negative = -1;

// when
final positive = Math.abc(number);

// then
assertThat(positive, -negative);
```

지금과 같이 시나리오 기반의 인수 테스트인 경우 주로 given은 특정 기능(지금과 같은 경우는 지하철 노선 생성 요청)을 수행하기 전에 선행되어야할 것을 뜻합니다!

> When Cucumber executes a Given step, it will configure the system to be in a well-defined state, such as creating and configuring objects or adding data to a test database.
https://cucumber.io/docs/gherkin/reference/#given

그래서 사실 제가 말씀드리고 싶었던 것은 인수 테스트에서 given의 의미를 떠올려 보셨으면 하는 생각에 의견드렸던 것이었습니다 :)

Reply:
네 ! 조언 감사합니다. BDD의 각각 행동에 대해 공부를 해봐야겠네요 .. !
링크 확인해보겠습니다 !